redbeams/src/main/java/com/sequenceiq/redbeams/service/dbserverconfig/DatabaseServerConfigService.java

```diff
@@ -40,7 +41,7 @@ public DatabaseServerConfig create(DatabaseServerConfig resource, Long workspace
             // prepareCreation(resource);
             resource.setWorkspaceId(workspaceId);
             return repository.save(resource);
-        } catch (AccessDeniedException e) {
+        } catch (AccessDeniedException | DataIntegrityViolationException e) {
             ConstraintViolationException cve = null;
             for (Throwable t = e.getCause(); t != null; t = t.getCause()) {
```

Comment:
Do you have access to Throwables.getCausalChain()?

Reply:
I'll check!

Reply:
Yep - I'll switch to using it.